src/ngraph_encapsulate_op.h

```diff
@@ -0,0 +1,114 @@
+/*******************************************************************************
+ * Copyright 2017-2019 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *******************************************************************************/
+
+#pragma once
+
+#ifndef NGRAPH_TF_ENCAPSULATE_OP_H_
+#define NGRAPH_TF_ENCAPSULATE_OP_H_
+
+#include <ostream>
+#include <vector>
+
+#include "ngraph/ngraph.hpp"
+
+#include "ngraph_freshness_tracker.h"
+
+#include "ngraph_log.h"
+
+#include "tensorflow/core/framework/tensor_shape.h"
+#include "tensorflow/core/graph/graph.h"
+
+namespace tensorflow {
+
+using NgFunctionIOCache = std::unordered_map<
+    std::shared_ptr<ngraph::runtime::Executable>,
+    std::vector<std::pair<void*, shared_ptr<ng::runtime::Tensor>>>>;
+
+namespace ngraph_bridge {
+
+class NGraphEncapsulateOp : public OpKernel {
+ public:
+  explicit NGraphEncapsulateOp(OpKernelConstruction* ctx);
+  ~NGraphEncapsulateOp() override;
+  Status ComputeSignature(std::vector<Tensor>& input_tensors,
+                          std::vector<TensorShape>& input_shapes,
+                          std::vector<const Tensor*>& static_input_map,
+                          std::stringstream& signature_ss);
+  Status GetNgExec(std::vector<Tensor>& input_tensors,
+                   std::pair<string, int64> ctx_params,
+                   std::vector<TensorShape>& input_shapes,
+                   std::vector<const Tensor*>& static_input_map,
+                   ng::runtime::Backend*& op_backend,
+                   std::shared_ptr<ngraph::runtime::Executable>& ng_exec);
+  Status AllocateTensorInput(
+      std::vector<Tensor>& input_tensors,
+      std::shared_ptr<ngraph::runtime::Executable>& ng_exec,
+      std::vector<TensorShape>& input_shapes, ng::runtime::Backend* op_backend,
```

Comment:
if possible, some of these inputs for the new functions can be marked `const T&` (from just `T&`)